shell/platform/android/io/flutter/view/FlutterView.java

```diff
@@ -533,8 +541,7 @@ public boolean onHoverEvent(MotionEvent event) {
 
         boolean handled = handleAccessibilityHoverEvent(event);
         if (!handled) {
-            // TODO(ianh): Expose hover events to the platform,
-            // implementing ADD, REMOVE, etc.
+            onTouchEvent(event);
```

Comment:
this doesn't seem to implement the intended protocol that we have designed for pointer events. The intended protocol is that we fire an "ADD" when we learn of the pointer, "MOVE" events while the pointer is moving, "DOWN" when it goes down/clicks, "UP" when it goes up, "MOVE" when it moves, and "REMOVE" when the pointer finally leaves the scene.

Reply:
see for instance https://master-docs-flutter-io.firebaseapp.com/flutter/gestures/PointerAddedEvent-class.html

Reply:
So this desired functionality is going to be taken care of in the framework, not in the engine. Like we agreed, we'll simply add the mouse with a special pointer index of 0 and not ever remove it.